Server/Components/ObjectOscillator.cpp

```diff
@@ -15,16 +16,21 @@ ObjectOscillator::ObjectOscillator(GameObject * go, float period, float distance
 
 void ObjectOscillator::OnCollisionEnter(GameObject* other) {
     if (other->GetTag() == PLAYER) {
-        if(other->GetTransform().GetPosition().y > gameObject->GetTransform().GetPosition().y)
-        objectsOnPlatform[other] = true;
-        std::cout << "PLAYER ON PLATFORM\n";
+
+        float topOfOscillator = gameObject->GetTransform().GetPosition().y + halfHeight;
+        CapsuleVolume* capVolume = ((CapsuleVolume*)other->GetBoundingVolume());
+        float playerFeet = other->GetTransform().GetPosition().y - capVolume->GetHalfHeight();
```

Comment:
Shouldn't this include the radiusof the capsule? Consdering the capsule as a line segment, to my understanding the half height is half the distance from a - b that then needs to be added together with the radius. Not 100% on this so get back to me about it.

Reply:
talked about this, getting this on main and if required will add on later. Small value anyway, shouldn't matter too much.